docs/distribution/README.md

```diff
@@ -0,0 +1,566 @@
+
+To generalize discovery, a `referrer-metadata` API is proposed to enable discovery of an referenced objects. To support this reverse lookup prototype, additions are proposed to the Notary v2 fork of the reference implementation [docker/distribution][notaryv2-distribution] through [notaryv2-referrer-api].
+
+A referrer is any registry artifact that has an immutable reference to a manifest. An OCI index is a referrer to each manifest it references. The [OCI image spec][oci-image] does not include a config property for an OCI index and there is no reverse lookup of referrers in docker distribution.
+
+A modified OCI index with an `index.config` property that references a collection of manifests allows us to associate a "type" to the referrer-referenced relationship, where the `index.config.mediaType` = `application/vnd.cncf.notary.config.v2+jwt`.
+
+### referrer-metadata API
+
+```HTTP
+`GET http://localhost:5000/v2/net-monitor/manifests/11wma2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49a11wm/referrer-metadata`
+```
+
+Using the diagram above, the `net-monitor:v1` manifest tag (4) has a digest of: `sha256:11wma2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49a11wm`. When requesting the referenced objects, we see 2 signature objects being returned (wabbit-networks (6) & acme-rockets(7)), and an OCI multi-arch index (1).
+
+The response could be in the following format. Note the additional `config-mediaType` to identify the specific artifact type in the results.
+
+```HTTP
+200 OK
+Content-Type: application/json
+{
+  "digest": "sha256:11wma2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49a11wm",
+  "references": [
+    {
+      "digest": "sha256:222mbbf80b44ce6be8234e6ff90a1ac34acbeb826903b02cfa0da11c82cb222m",
+      "mediaType": "application/vnd.oci.image.index.v1+json",
+      "size": "1024",
+      "config-mediaType": "application/vnd.cncf.notary.config.v2+jwt"
+    },
+    {
+      "digest": "sha256:333mc0c33ebc4a74a0a554c86ac2b28ddf3454a5ad9cf90ea8cea9f9e75c333m",
+      "mediaType": "application/vnd.oci.image.index.v1+json",
+      "size": "1025",
+      "config-mediaType": "application/vnd.cncf.notary.config.v2+jwt"
+    },
+    {
+      "digest": "sha256:111ia2d22ae5ef400769fa51c84717264cd1520ac8d93dc071374c1be49a111i",
+      "mediaType": "application/vnd.oci.image.index.v1+json",
+      "size": "1025",
+      "config-mediaType": "application/vnd.oci.image.index.v1+json"
+    }
+  ]
+}
+```
+
+## Persisting Referrer Metadata (Signatures)
+
+The proposal implements a referrer metadata store for manifests that is essentially a reverse-lookup, by `mediaType`, to referrer config objects. For example, when an OCI index is pushed, if it references a config object of media type `application/vnd.cncf.notary.config.v2+jwt`, a link to the config object is recorded in the referrer metadata store of each referenced manifest.

```

Comment:
How do you generate the response in section ` referrer-metadata API` with such data structure?  IMO the linkage is between the referrer and the manifest, rather than between the config object of the referrer and the manifest.

IMO instead of only storing the referrer metadata, you should first store the referrer?

Reply:
/cc @SteveLasker to comment on the requirement for including complete descriptors in the response

@reasonerjt , you're right. The linkage is between the referrer and the manifest. However, with this approach, we get some optimization for referrer metadata lookups since clients would ask for referrer config objects when requesting signatures. What scenarios would require a client to query the list of referrers for a manifest?